File: cases/taint/cpp/exception-catch-negative/exception_catch_negative.cpp
struct FlowException {
    int value;
};

int dfb_source() { // DFB-SOURCE: exception-catch-input
    return 1;
}

void dfb_sink(int value) {} // DFB-SINK: exception-catch-sink

void run() {
    try {
        FlowException flow;
        int ignored = dfb_source();
        flow.value = 0;
        throw flow; // DFB-WITNESS: exception-catch-throw
    } catch (FlowException &caught) {
        dfb_sink(caught.value);
    }
}

File: cases/taint/php/exception-catch-negative/exception_catch_negative.php
<?php
class FlowException extends \Exception
{
    public string $value = "clean";
}

function dfb_source(): string { // DFB-SOURCE: exception-catch-input
    return "tainted";
}

function dfb_sink(string $value): void {} // DFB-SINK: exception-catch-sink

function run(): void {
    try {
        $flow = new FlowException("flow");
        $ignored = dfb_source();
        $flow->value = "clean";
        throw $flow; // DFB-WITNESS: exception-catch-throw
    } catch (FlowException $caught) {
        dfb_sink($caught->value);
    }
}
